### PR TITLE
fix: request loop slow sync - WPB-6502

### PR DESCRIPTION
**Pull request by netbe**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6502" title="WPB-6502" target="_blank">StoryWPB-6502</a>  [iOS] request loop on /notifications and app stuck in sync
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

During the 3.111.X releases, a lot of users had database crashes and resulted in clearing the database. Doing that they will start fresh but with a persisted lastEventId in the userDefaults. So we'll not get a correct lastEventId for their new client.

### Causes (Optional)

We started the slow sync with the wrong case: fetchingMissingEvents (`/notifications) with an incorrect lastEventId.

if /notifications returns a 404, we'll force a slowSync and clearing the lastEventId. Unfortunately this forceSlowSync was starting on the wrong phase `.fetchingMissingEvents` instead of `.fetchingLastUpdateEventID` (`/notifications/last`). So we'll end up doing again (request loop) a `/notifications` returning a 404, and triggering a slowSync.

### Solutions

Start the slowSync with `.fetchingLastUpdateEventID`

In some cases, we have (old) database patches where a slowSync was triggered in order to sync resources, this behavior has been kept and moved to own method.
#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

* Insert duplicate user from previous build
* Upgrade, this will trigger a slow sync after the migration
There should not be a request loop

### Notes (Optional)

The database patches could be removed, but this can be done in a second time to clean up.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-sync-engine/Source/UserSession/SyncStatus.swift [hunk 1]
```diff
@@ -83,8 +83,8 @@ extension Notification.Name {
         currentSyncPhase = hasPersistedLastEventID ? .fetchingMissedEvents : .fetchingLastUpdateEventID
         notifySyncPhaseDidStart()
 
-        self.forceSlowSyncToken = NotificationInContext.addObserver(name: .ForceSlowSync, context: managedObjectContext.notificationContext) { [weak self] (_) in
-            self?.forceSlowSync()
+        self.forceSlowSyncToken = NotificationInContext.addObserver(name: .resyncResources, context: managedObjectContext.notificationContext) { [weak self] (_) in
```
Comment (johnxnguyen) [inline wire-ios-sync-engine/Source/UserSession/SyncStatus.swift]: Should we also rename the token?

File: wire-ios-sync-engine/Source/UserSession/SyncStatus.swift [hunk 2]
```diff
@@ -110,12 +110,22 @@ extension Notification.Name {
...
         syncStateDelegate.didStartSlowSync()
     }
 
+    /// Sync the resources: Teams, Users, Conversations...
+    func resyncResources() {
+       // Refetch user settings.
+       ZMUser.selfUser(in: managedObjectContext).needsPropertiesUpdate = true
+       // Set the status.
+       currentSyncPhase = SyncPhase.fetchingLastUpdateEventID.nextPhase
+       self.log("slow sync")
```
Comment (johnxnguyen) [inline wire-ios-sync-engine/Source/UserSession/SyncStatus.swift]: Should we also log "resync resources"?

**Review by johnxnguyen — APPROVED** [review]

File: wire-ios-sync-engine/Source/UserSession/SyncStatus.swift (hunk 1 above)
Comment (netbe) [reply wire-ios-sync-engine/Source/UserSession/SyncStatus.swift]: yep

File: wire-ios-sync-engine/Source/UserSession/SyncStatus.swift (hunk 2 above)
Comment (netbe) [reply wire-ios-sync-engine/Source/UserSession/SyncStatus.swift]: make sense

**Review by agisilaos — APPROVED** [review]

**Review by typfel — APPROVED** [review]

**Pull request merged**